### DISCUSSION TITLE
How to filter different hierarchies with single filter?

### DISCUSSION THREAD

**Discussion by ihernandezvelarde** [atoti/atoti#753] [Q&A]
Good morning!

I would like to know how to filter two different dashboards with a single filter, when the information contained in the two hierarchies are the same. In this case we can see that table1 contains month and table2 contains month2, both contain month, but if we filter by month, it only filters by one of them.

The expected result would be that with a single "quick filter", we could filter both tables without having to filter first one month and then the other with month2.

## Steps to reproduce
We have Table1 with:
-  ID, TYPE, ID2, MONTH y VALUE

And Table2 with:
- ID, MONTH2 y VALUE2

![image](https://user-images.githubusercontent.com/91269912/219004419-15f2e98f-4108-4de8-b967-27ffaf83ca75.png)

The tables contain these inputs:

id,type,id2,month,value          
1,type1,,1,1
2,type1,,3,2
3,type2,,5,3
4,type2,,7,4
5,type3,1,,,
6,type3,2,,, 

id,month2,value2
1,1,1
2,2,2

## Actual Result
In this image we have on the right the information of table 1 and on the left the information of table 2, below a filter by the field month1 of table1.
![image](https://user-images.githubusercontent.com/91269912/219004559-927f8f85-8ff9-4ff1-9a7d-8cd234dc9d51.png)

If we select one of these months, only the information from table1 is displayed.
![image](https://user-images.githubusercontent.com/91269912/219004659-ae38c220-c8bb-4906-a2ee-4f4ba2b6352b.png)

In this image we have the same but the quick filter is by month2, so it is only showing us the information of table2.
![image](https://user-images.githubusercontent.com/91269912/219006989-24dbefe9-a362-4153-980e-805557987347.png)

## Expected Result

The expected result is to be able to filter both tables with a single "quick filter", without having to filter first one month to get the information from the first table and then having to do a second filter to get the information from the second table.

In the previous case, we would like to see that when applying month = 1, the two tables are updated with month and month2 = 1.

## Environment

- atoti: 0.7.2
- Python: 3.9.13
- Operating System: Windows 10 Enterprise

**Comment by tibdex** [comment]
Hi,

A filter can only be applied to a single hierarchy. Can you make this change:

```diff
- table1.join(table2, mapping={"ID2": "ID"})
+ table1.join(table2, mapping={"ID2": "ID", "MONTH": "MONTH2"})
```

?

This would leave you with a single **MONTH** hierarchy that would behave as you expect.

**Reply by ihernandezvelarde** [reply]
Hi,

If we put month with month2, when you filter by month, the ID2 comes out as N/A and you only get the type 1 and not the type 3.

![image](https://user-images.githubusercontent.com/91269912/219312330-d239ebd6-ae41-4afc-bb34-2ac61e3adad2.png)

Thanks for the reply!

**Reply by tibdex** [reply]
Yes that's due to the `N/A`s in `TABLE 1`'s `ID2` and `MONTH` columns which prevent some of `TABLE 1`'s facts/rows to be joined with `TABLE 2`. Can you process the data in a way to get rid of these `N/A`?
